### PR TITLE
iam_cert return arn and allow use with ansible vault

### PR DESCRIPTION
**Pull request by wrouesnel**
This is a small set of changes which make the iam_cert module much more useful with AWS by allowing certificates to be specified inline as parameters (i.e. via ansible-vault variables) and by returning the certificate ARN in all success cases, which allows it to be directly used to provision elastic load balancers.

There is also an included fix to allow duplicates to be correctly detected when the certificate body includes the authority chain directly (the data returned by AWS previously did not match).

### CODE REVIEW

File: cloud/amazon/iam_cert.py [hunk 1]
```diff
@@ -105,9 +104,9 @@
 try:
     import boto
     import boto.iam
-    HAS_BOTO = True
 except ImportError:
-    HAS_BOTO = False
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
```
Comment (marvinpinto) [inline cloud/amazon/iam_cert.py]: You should probably use something like this here instead:

```
fail_json(changed=False, msg='Python boto is required for this module')
```

http://docs.ansible.com/ansible/developing_modules.html#module-checklist

**Comment by wrouesnel** [comment]
Rebased against latest devel head.

**Comment by gregdek** [comment]
Thanks @wrouesnel. @defionscode please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
Thanks @wrouesnel. @defionscode please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by robynbergeron** [comment]
@defionscode This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by robynbergeron** [comment]
Thanks @wrouesnel for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by gregdek** [comment]
@wrouesnel A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by wrouesnel** [comment]
Ah, right - I will get right on that!

**Comment by wrouesnel** [comment]
Ok, rebased.

**Comment by gregdek** [comment]
Thanks @wrouesnel. @defionscode please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by juliedavila** [comment]
Unfortunately I do not have time to review PRs at this time. @gregdek we'll need others to review and approve this.

**Comment by phumpal** [comment]
:+1: **worksforme**

@wrouesnel - Changes LGTM. I was also able to create an `iam_cert` and assign it to an ELB.

@gregdek @robynbergeron - Can either of you PTAL again? e.g. `shipit` :)

**Comment by gregdek** [comment]
Thanks @wrouesnel for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

**Comment by juliedavila** [comment]
shipit, does the same thing as #2843, I prefer for the first submitter to get the change merged in

**Comment by elnur** [comment]
I hope it doesn't take another half a year to get this merged. :wink:

**Comment by gregdek** [comment]
Thanks again to @wrouesnel for this PR, and thanks @ansible for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by the2hill** [comment]
Any updates?

**Pull request closed without merging**